hack/make-rules/verify/shellcheck.sh

```diff
@@ -53,7 +68,18 @@ SHELLCHECK_OPTIONS=(
 )
 
 # actually shellcheck
-docker run \
-  --rm -t -v "${REPO_ROOT}:${REPO_ROOT}" -w "${REPO_ROOT}" \
-  "${SHELLCHECK_IMAGE}" \
-  "${SHELLCHECK_OPTIONS[@]}" "${all_shell_scripts[@]}"
+# tell the user which we've selected and lint all scripts
+# The shellcheck errors are printed to stdout by default, hence they need to be redirected
+# to stderr in order to be well parsed for Junit representation by juLog function
+res=0
+if ${HAVE_SHELLCHECK}; then
+  >&2 echo "Using host shellcheck ${SHELLCHECK_VERSION} binary."
+  shellcheck "${SHELLCHECK_OPTIONS[@]}" "${all_shell_scripts[@]}" >&2 || res=$?
+else
+  >&2 echo "Using shellcheck ${SHELLCHECK_VERSION} docker image."
+  "${DOCKER}" run \
+    --rm -v "${KUBE_ROOT}:${KUBE_ROOT}" -w "${KUBE_ROOT}" \
+    "${SHELLCHECK_IMAGE}" \
+  shellcheck "${SHELLCHECK_OPTIONS[@]}" "${all_shell_scripts[@]}" >&2 || res=$?
```

Comment:
you didn't use `shellheck` before , it was directly `"${SHELLCHECK_OPTIONS[@]}" "${all_shell_scripts[@]}"`

Reply:
Well you can check that this works correctly in CI

Reply:
the entrypoint is different between `koalaman/shellcheck:v0.7.1` and `koalaman/shellcheck-alpine:v0.8.0` (alpine image has none)

```
$ podman pull docker.io/koalaman/shellcheck:v0.7.1
Trying to pull docker.io/koalaman/shellcheck:v0.7.1...
Getting image source signatures
Copying blob b41a66f4a642 done
Copying blob 3ddc77a5eea0 done
Copying config 3d5a3cb1aa done
Writing manifest to image destination
Storing signatures
3d5a3cb1aa471d460574334cc50692a7b8ddb2ab2d3d075e99eebdc964dc4445

$ podman inspect 3d5a3cb1aa471d460574334cc50692a7b8ddb2ab2d3d075e99eebdc964dc4445
...
               "Entrypoint": [
                    "/bin/shellcheck"
               ],
...

$ podman pull docker.io/koalaman/shellcheck-alpine:v0.8.0@sha256:f42fde76d2d14a645a848826e54a4d650150e151d9c81057c898da89a82c8a56
Trying to pull docker.io/koalaman/shellcheck-alpine@sha256:f42fde76d2d14a645a848826e54a4d650150e151d9c81057c898da89a82c8a56...
Getting image source signatures
Copying blob 9d2ca41a6df5 done
Copying blob a0d0a0d46f8b done
Copying config bc3f42c93d done
Writing manifest to image destination
Storing signatures
bc3f42c93dab860e166a3ea21d913d73051240610b628a72716ce221354bf3bb

$ podman inspect bc3f42c93dab860e166a3ea21d913d73051240610b628a72716ce221354bf3bb
...
               "Cmd": [
                    "/bin/sh"
               ],
...
```

Reply:
thanks for clarify